src/Utilities/PointerVector.hpp

```diff
@@ -23,23 +23,56 @@
 #define SPECTRE_BLAZE_DEALLOCATOR blaze::ArrayDelete()
 
 namespace blaze {
+
 template <typename T>
 BLAZE_ALWAYS_INLINE SIMDdouble step_function(const SIMDf64<T>& v) noexcept
 #if BLAZE_AVX512F_MODE || BLAZE_MIC_MODE
 {
-  return _mm512_set_pd((~v)[7] < 0.0 ? 0.0 : 1.0, (~v)[6] < 0.0 ? 0.0 : 1.0,
-                       (~v)[5] < 0.0 ? 0.0 : 1.0, (~v)[4] < 0.0 ? 0.0 : 1.0,
-                       (~v)[3] < 0.0 ? 0.0 : 1.0, (~v)[2] < 0.0 ? 0.0 : 1.0,
-                       (~v)[1] < 0.0 ? 0.0 : 1.0, (~v)[0] < 0.0 ? 0.0 : 1.0);
+  // A step function entirely in terms of intel vectorization instructions
+  // first, we `reinterpret_cast` to an integer type, as the bit operations
```

Comment:
first -> First

Reply:
also: period at end of preceding sentence.